source/Politics.cpp

```diff
@@ -229,6 +226,16 @@ string Politics::Fine(PlayerInfo &player, const Government *gov, int scan, const
 
 		if(!scan || (scan & ShipEvent::SCAN_CARGO))
 		{
+			// Illegal goods can be hidden inside legal goods to avoid detection.
+			const int contraband = ship->Cargo().IllegalCargoAmount();
+			const int legalGoods = ship->Cargo().Used() - contraband;
+			const int netIllegalCargo = contraband - ship->Attributes().Get("scan concealment");
+
+			const double illegalRatio = legalGoods ? max(1., 2. * netIllegalCargo / legalGoods) : 1.;
```

Comment:
So, if there's no legal cargo, then the scan concealment has no effect? Nor does the amount or brightness of illegal cargo.

Reply:
Oh, yeah. This should probably be something like 
```c++
const double illegalRatio = legalGoods ? max(1., 2. * netIllegalCargo / legalGoods) : (netIllegalCargo > 0 ? 1. : 0.);
```

Reply:
> Nor does the amount or brightness of illegal cargo.

The scan brightness is considered by IllegalCargoAmount, which is what is used here.